docs/understand/architecture.md

```diff
@@ -57,6 +72,35 @@ The credential scheme, along with the request signing logic are compatible with
 
 Currently, the auth service manages its own database of users and credentials and does not use IAM in any way. 
 
+### Hooks Engine
+
+The hooks engine enables CI/CD for data by triggering user defined [Actions](../setup/hooks.md) that will run during commit/merge. 
+
 ### Frontend UI
 
 The UI layer is a simple browser-based client that uses the OpenAPI server. It allows management, exploration and data access to repositories, branches, commits and objects in the system.
+
+## lakeFS Clients
+
+There are several ways of interacting with lakeFS. Many data applications have different use-cases for lakeFS and 
+we try to support as many as possible.
+
+### OpenAPI Generated SDKs
```

Comment:
Can you move all the clients up here so that they are listed after the lakefS clients title?

Reply:
I prefer to keep the sdks first. They are clients too, and are used by the other clients as well.